docs/openapi3.0.yml

```diff
@@ -19,35 +19,56 @@ components:
           type: string
         type: 
           type: string
-          enum: ["user", "pro", "ai"]       
-    Question:
+          enum: ["user", "pro", "ai"]
+
+    UserContent:
       type: object
       properties:
-        id: 
-          type: string
-        title: 
+        id:
           type: string
-        content: 
+        content:
           type: string
-        tags: 
-          type: array
-          items:
-            type: string
-        likes: 
+        likes:
           type: integer
         dislikes:
           type: integer
-        answers:
-          type: integer
         created:
           type: string
           description: The date, the question was created
-        updated: 
+        rating:
           type: string
-          description: The date, the question was last updated
-        author: 
-           $ref: '#/components/schemas/Author'
-            
+          enum:
+            - like
+            - dislike
+            - none
+          
+    Question:
+      allOf:
+        - $ref: '#/components/schemas/UserContent'
+        - type: object
+          properties:
+            title:
+              type: string
+            tags: 
+              type: array
+              items:
+                type: string
+            answers:
+              type: integer
+            updated: 
+              type: string
+              description: The date, the question was last updated
+            author: 
+                $ref: '#/components/schemas/Author'
+
+    Answer:
+      allOf:
+        - $ref: '#/components/schemas/UserContent'
+        - type: object
+          properties:
+            author:
```

Comment:
add enum typeOfAI with [user, wolframAlpha, chatGPT, ...]

Reply:
I hope this solves the request
```
...   
         author:
              oneOf: [
                {
                  $ref:  '#/components/schemas/Author'
                },
                {
                  $ref: '#/components/schemas/AItype'
                }
              ]
    AItype:
      type: object
      properties:
        type:
          aiType: string
          enum:
            - WolframAlpha
            - GPT
...
```

Reply:
use name and type of author to realize an AI Author